inlong-audit/audit-source/bin/flume-ng

```diff
@@ -0,0 +1,379 @@
+#LD_LIBRARY_PATH=""
+
+opt_conf=""
+opt_classpath=""
+opt_dryrun=""
+
+mode=$1
+shift
+
+case "$mode" in
+  help)
+    display_help
+    exit 0
+    ;;
+  agent)
+    opt_agent=1
+    ;;
+  node)
+    opt_agent=1
+    warn "The \"node\" command is deprecated. Please use \"agent\" instead."
+    ;;
+  avro-client)
+    opt_avro_client=1
+    ;;
+  version)
+   opt_version=1
+   CLEAN_FLAG=0
+   ;;
+  *)
+    error "Unknown or unspecified command '$mode'"
+    echo
+    display_help
+    exit 1
+    ;;
+esac
+
+while [ -n "$*" ] ; do
+  arg=$1
+  shift
+
+  case "$arg" in
+    --conf|-c)
+      [ -n "$1" ] || error "Option --conf requires an argument" 1
+      opt_conf=$1
+      shift
+      ;;
+    --classpath|-C)
+      [ -n "$1" ] || error "Option --classpath requires an argument" 1
+      opt_classpath=$1
+      shift
+      ;;
+    --dryrun|-d)
+      opt_dryrun="1"
+      ;;
+    -D*)
+      opt_java_props="$opt_java_props $arg"
+      ;;
+    -X*)
+      opt_java_props="$opt_java_props $arg"
+      ;;
+    *)
+      args="$args $arg"
+      ;;
+  esac
+done
+
+# make opt_conf absolute
+if [[ -n "$opt_conf" && -d "$opt_conf" ]]; then
+  opt_conf=$(cd $opt_conf; pwd)
+fi
+
+# allow users to override the default env vars via conf/flume-env.sh
+if [ -z "$opt_conf" ]; then
+  warn "No configuration directory set! Use --conf <dir> to override."
+elif [ -f "$opt_conf/flume-env.sh" ]; then
+  info "Sourcing environment configuration script $opt_conf/flume-env.sh"
+  source "$opt_conf/flume-env.sh"
+fi
+
+# append command-line java options to stock or env script JAVA_OPTS
+if [ -n "${opt_java_props}" ]; then
+  JAVA_OPTS="${JAVA_OPTS} ${opt_java_props}"
+fi
+
+# prepend command-line classpath to env script classpath
+if [ -n "${opt_classpath}" ]; then
+  if [ -n "${FLUME_CLASSPATH}" ]; then
+    FLUME_CLASSPATH="${opt_classpath}:${FLUME_CLASSPATH}"
+  else
+    FLUME_CLASSPATH="${opt_classpath}"
+  fi
+fi
+
+if [ -z "${FLUME_HOME}" ]; then
+  FLUME_HOME=$(cd $(dirname $0)/..; pwd)
+fi
+
+# prepend $FLUME_HOME/lib jars to the specified classpath (if any)
+if [ -n "${FLUME_CLASSPATH}" ] ; then
+  FLUME_CLASSPATH="${FLUME_HOME}/lib/*:$FLUME_CLASSPATH"
+else
+  FLUME_CLASSPATH="${FLUME_HOME}/lib/*"
+fi
+
+# find java
+if [ -z "${JAVA_HOME}" ] ; then
+  warn "JAVA_HOME is not set!"
+  # Try to use Bigtop to autodetect JAVA_HOME if it's available
+  if [ -e /usr/libexec/bigtop-detect-javahome ] ; then
+    . /usr/libexec/bigtop-detect-javahome
+  elif [ -e /usr/lib/bigtop-utils/bigtop-detect-javahome ] ; then
+    . /usr/lib/bigtop-utils/bigtop-detect-javahome
+  fi
+
+  # Using java from path if bigtop is not installed or couldn't find it
+  if [ -z "${JAVA_HOME}" ] ; then
+    JAVA_DEFAULT=$(type -p java)
+    [ -n "$JAVA_DEFAULT" ] || error "Unable to find java executable. Is it in your PATH?" 1
+    JAVA_HOME=$(cd $(dirname $JAVA_DEFAULT)/..; pwd)
+  fi
+fi
+
+# look for hadoop libs
+#add_hadoop_paths
+#add_HBASE_paths
```

Comment:
`hadoop` will not be used in ` audit module`, it's better to remove these comments.

Reply:
ok